gdal/apps/gdal_grid_lib.cpp

```diff
@@ -823,13 +832,27 @@ GDALDatasetH GDALGrid( const char *pszDest, GDALDatasetH hSrcDataset,
     if ( psOptions->pszSQL )
         nBands++;
 
-    // FIXME
-    int nXSize = psOptions->nXSize;
-    if ( nXSize == 0 )
-        nXSize = 256;
-    int nYSize = psOptions->nYSize;
-    if ( nYSize == 0 )
-        nYSize = 256;
+    int nXSize;
+    int nYSize;
+    if ( psOptions->dfXRes && psOptions->dfYRes )
+    {
+        nXSize = static_cast<int>(
```

Comment:
This also assumes that -txe and -tye have been specified. Making the code to work when they are not specified would require more changes, so if you don't want to proceed to them, you should at least test that the bounds have been set, and error out if not. And document that restriction in the .rst

Reply:
Added in 7ad261a and 18ef73c